### PR TITLE
Add the test cases to test_default.py

### PR DESCRIPTION
**Pull request by hkx303**
**Description**
- [x] test_guest_attr_by_curl
- [x] test_associated_info_by_rhsmlog_and_webui
- [x] test_mapping_info
- [x] test_guest_facts 
- [x] test_delete_host_hypervisor

**Test Result**
```
[hkx303@kuhuang virtwho-test]$ pytest tests/hypervisor/test_libvirt.py -k test_check_guest_attr_by_curl -s
==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-5.4.3, py-1.8.0, pluggy-0.13.1
rootdir: /home/hkx303/Documents/CI/virtwho-test, inifile: pytest.ini
plugins: services-1.3.1, mock-1.10.4, forked-1.0.2, ibutsu-1.0.34, cov-2.7.1, xdist-1.34.0
collected 13 items / 12 deselected / 1 selected   
=================== 1 passed, 12 deselected, 1 warning in 89.15s (0:01:29) ===================
```
```
[hkx303@kuhuang virtwho-test]$ pytest tests/hypervisor/test_libvirt.py -k test_associated_info_by_rhsmlog_and_webui -s
==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-5.4.3, py-1.8.0, pluggy-0.13.1
rootdir: /home/hkx303/Documents/CI/virtwho-test, inifile: pytest.ini
plugins: services-1.3.1, mock-1.10.4, forked-1.0.2, ibutsu-1.0.34, cov-2.7.1, xdist-1.34.0
collected 14 items / 13 deselected / 1 selected  
================== 1 passed, 13 deselected, 2 warnings in 64.48s (0:01:04) ===================
```
```
[hkx303@kuhuang virtwho-test]$ pytest tests/hypervisor/test_default.py -k test_mapping_info -s
==================================== test session starts =====================================

platform linux -- Python 3.7.3, pytest-5.4.3, py-1.8.0, pluggy-0.13.1

rootdir: /home/hkx303/Documents/CI/virtwho-test, inifile: pytest.ini

plugins: services-1.3.1, mock-1.10.4, forked-1.0.2, ibutsu-1.0.34, cov-2.7.1, xdist-1.34.0

collected 5 items / 4 deselected / 1 selected 
========================= 1 passed, 4 deselected in 93.70s (0:01:33) =========================
```  

```
[hkx303@kuhuang virtwho-test]$ pytest tests/hypervisor/test_default.py -k test_guest_facts
==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-5.4.3, py-1.8.0, pluggy-0.13.1
rootdir: /home/hkx303/Documents/CI/virtwho-test, inifile: pytest.ini
plugins: services-1.3.1, mock-1.10.4, forked-1.0.2, ibutsu-1.0.34, cov-2.7.1, xdist-1.34.0
collected 4 items / 3 deselected / 1 selected      
============================== 1 passed, 3 deselected in 29.17s ==============================
```
```
[hkx303@kuhuang virtwho-test]$ pytest tests/hypervisor/test_default.py -k test_delete_host_hypervisor -s
==================================== test session starts =====================================
platform linux -- Python 3.7.3, pytest-5.4.3, py-1.8.0, pluggy-0.13.1
rootdir: /home/hkx303/Documents/CI/virtwho-test, inifile: pytest.ini
plugins: services-1.3.1, mock-1.10.4, forked-1.0.2, ibutsu-1.0.34, cov-2.7.1, xdist-1.34.0
collected 5 items / 4 deselected / 1 selected   
======================== 1 passed, 4 deselected in 214.44s (0:03:34) =========================
```

### CODE REVIEW

File: tests/hypervisor/test_default.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+        # run virt-who to send mappings
+        result = virtwho.run_service()
+        assert (result['error'] == 0
+                and result['send'] == 1
+                and result['thread'] == 1)
+
+        # delete virt-who host from webui
+        if REGISTER == 'rhsm':
+            rhsm.delete(virtwho_hostname)
```
Comment (Junefen) [inline tests/hypervisor/test_default.py:208]: which should be changed to rhsm.host_delete

File: tests/hypervisor/test_default.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        else:
+            satellite.host_delete(virtwho_hostname)
+
+        result = virtwho.run_service()
+        assert (result['error'] == 0
```
Comment (Junefen) [inline tests/hypervisor/test_default.py:213]: as my understanding, at the first we should configure virt-who conf file without any rhsm_ options, then remove the virt-who host from server side, then run virt-who again, which should fail to report. and then register the vrt-who host again.

File: tests/hypervisor/test_default.py (hunk 1 above)
Comment (hkx303) [reply tests/hypervisor/test_default.py:208]: Ok..., I just re-tested this after the rebase

File: tests/hypervisor/test_default.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+from virtwho import REGISTER
+from virtwho import HYPERVISOR
+
+from virtwho.base import hostname_get
+
+
+@pytest.mark.usefixtures('function_virtwho_d_conf_clean')
+@pytest.mark.usefixtures('class_debug_true')
+@pytest.mark.usefixtures('class_globalconf_clean')
+class TestLibvrtPositive:
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/hypervisor/test_default.py:18]: updated

File: tests/hypervisor/test_default.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            ret, output = ssh_host.runcmd(cmd)
+            attr1 = f'"id":{guest_registered_id}'
+            assert (attr1 in output
+                    and guest_hostname in output)
+
+    @pytest.mark.tier1
+    def test_associated_info_by_rhsmlog_and_webui(
+            self, virtwho, function_hypervisor, hypervisor_data, rhsm, satellite):
+        """
+        :title: virt-who: libvirt: check associated info by rhsm.log and webui
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/hypervisor/test_default.py:64]: updated

File: tests/hypervisor/test_default.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+        if REGISTER == 'rhsm':
+            assert rhsm.associate(host_name, guest_uuid)
+        else:
+            assert satellite.associate_on_webui(host_name, guest_hostname)
+
+    @pytest.mark.tier1
+    def test_mapping_info(
+            self, virtwho, function_hypervisor, hypervisor_data, rhsm, satellite, register_data):
+        """
+        :title: virt-who: default: test the mapping info
```
[the remark this answers is not in the document]
Comment (hkx303) [reply tests/hypervisor/test_default.py:95]: Great

File: tests/hypervisor/test_default.py (hunk 2 above)
Comment (hkx303) [reply tests/hypervisor/test_default.py:213]: Yep, we discussed this, forgot to update, my fault :/

**Pull request merged**